core/transactions/attempt_context_impl.cxx

```diff
@@ -1127,22 +1130,22 @@ attempt_context_impl::do_core_query(const std::string& statement,
     return f.get();
 }
 
-std::pair<couchbase::transaction_op_error_context, couchbase::transactions::transaction_query_result>
+std::pair<couchbase::error, couchbase::transactions::transaction_query_result>
 attempt_context_impl::do_public_query(const std::string& statement,
                                       const couchbase::transactions::transaction_query_options& opts,
                                       std::optional<std::string> query_context)
 {
     try {
         auto result = do_core_query(statement, opts, query_context);
-        return core::impl::build_transaction_query_result(result);
+        auto [ctx, res] = core::impl::build_transaction_query_result(result);
+        return std::make_pair(core::impl::make_error(ctx), res);
     } catch (const transaction_operation_failed& e) {
-        return { e.get_error_ctx(), {} };
+        return { core::impl::make_error(e), {} };
     } catch (const op_exception& qe) {
-        return { qe.ctx(), {} };
+        return { core::impl::make_error(qe.ctx()), {} };
```

Comment:
Not sure if maybe we should create a `make_error` for the op_exception (which contain the transaction_op_error_context and a cause), as right now just using the err context loses the `cause` field.

Reply:
It looks like that might not provide any additional information to what's already in the `transaction_op_error_context`, so I think it's fine as is. Considering we were only returning `transaction_op_error_context` in the Public API previously, this shouldn't lose any information that we used to expose.